### ISSUE TITLE
Add headers for CORS to support amp-ad on same origin

### ISSUE THREAD

**Issue by davisshaver** [ampproject/amp-wp#2226]
Not sure this is the right repo, but I have run into an issue with CORS using `<amp-ad>` and I am wondering if there's anything we can do in WordPress to help with this.

When trying to us a Doubleclick unit on a WordPress site, I get an error like this:

```
Access to fetch at 'https://securepubads.g.doubleclick.net/gampad/ads?adk=1703325866&sz=320x50&output=html&impl=ifr&ifi=3&adf=3000315708&nhd=0&eid=21060933&adx=248&ady=3979&oid=2&adsid=AGt39rQFFa4zk7Pz0p_D93sAvgqPkNP7oeTUElvnyxq3rqc5LeV2GkU3NA334PWkVCaQ&jar=2019-04-30-21&gdfp_req=1&sfv=1-0-23&u_sd=2&is_amp=3&amp_v=1904262212040&d_imp=1&c=749312000117&ga_cid=GA1.2.1773748353.1556658300&ga_hid=117&dt=1556660685990&biw=816&bih=831&u_aw=816&u_ah=831&u_cd=24&u_w=816&u_h=831&u_tz=-240&u_his=7&vis=1&scr_x=0&scr_y=0&bc=7&url=https%3A%2F%2Fonwardstate.com%2F2019%2F04%2F29%2Flooking-up-from-my-shoes-gabriela-stevensons-senior-column%2F&loc=https%3A%2F%2Fonwardstate.com%2F2019%2F04%2F29%2Flooking-up-from-my-shoes-gabriela-stevensons-senior-column%2Famp%2F%3Fcache-asdf&bdt=315&dtd=116&__amp_source_origin=https%3A%2F%2Fonwardstate.com'
from origin 'https://onwardstate.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

This doesn't really make sense to me as the requested resource is on the Google server. I raised this issue previously in the AMP repo but the solution wasn't clear to me. I am trying to get the AMP ads to work on both the self-hosted and cached version of the pages. 

Here is an example link: https://onwardstate.com/2019/04/29/looking-up-from-my-shoes-gabriela-stevensons-senior-column/amp/

**Comment by westonruter** [comment]
I don't think that this is an issue on your site. The problem I believe the problem is the request to `securepubads.g.doubleclick.net` is not sending the required `Access-Control-Allow-Origin` response header. Perhaps you have to whitelist `onwardstate.com` in your DoubleClick account?

**Comment by davisshaver** [comment]
@westonruter Thank you for taking a look. I don't see anything of that nature in the Ad Manager UI. I am really stumped on this one. I am going to try and open it as a new issue on the main AMP repo.

**Comment by davisshaver** [comment]
I was using `amp-ad-latest` and switching to `https://cdn.ampproject.org/v0/amp-ad-0.1.js` helped.

**Comment by westonruter** [comment]
That doesn't make sense because the two JS files should be the same.

**Comment by davisshaver** [comment]
@westonruter Thank you, good point. I will try to figure out what changed. It could have been an intermittent upstream issue. I believe it was the same ad tag being used and I'll double check that too. Closing for now though as at this point I don't think it was a WP-related issue.
